opm/input/eclipse/EclipseState/Grid/FieldProps.hpp

```diff
@@ -410,102 +414,114 @@ class FieldProps {
+        if (field_data.valid() || allow_unsupported) {
+            // Note: FieldDataManager depends on init_get<>() producing a
+            // long-lived FieldData instance.
+            return { keyword, GetStatus::OK, &field_data };
+        }
 
-        if (!has0) {
+        if (! has0) {
             this->erase<T>(keyword);
-            return FieldDataManager<T>(keyword, GetStatus::MISSING_KEYWORD, nullptr);
+
+            return { keyword, GetStatus::MISSING_KEYWORD, nullptr };
         }
 
-        return FieldDataManager<T>(keyword, GetStatus::INVALID_DATA, nullptr);
+        return { keyword, GetStatus::INVALID_DATA, nullptr };
     }
 
-
     template <typename T>
-    const std::vector<T>& get(const std::string& keyword) {
-        const auto& data = this->try_get<T>(keyword);
-        return data.data();
+    const std::vector<T>& get(const std::string& keyword)
+    {
+        return this->try_get<T>(keyword).data();
     }
 
     template <typename T>
-    std::vector<T> get_global(const std::string& keyword) {
-        const auto& managed_field_data = this->try_get<T>(keyword);
+    std::vector<T> get_global(const std::string& keyword)
+    {
+        const auto managed_field_data = this->try_get<T>(keyword);
         const auto& field_data = managed_field_data.field_data();
+
         const auto& kw_info = Fieldprops::keywords::global_kw_info<T>(keyword);
-        if (kw_info.global)
-            return *field_data.global_data;
-        else
-            return this->global_copy(field_data.data, kw_info.scalar_init);
-    }
 
+        return kw_info.global
+            ? *field_data.global_data
+            : this->global_copy(field_data.data, kw_info.scalar_init);
+    }
 
     template <typename T>
-    std::vector<T> get_copy(const std::string& keyword, bool global) {
-        bool has0 = this->has<T>(keyword);
+    std::vector<T> get_copy(const std::string& keyword, bool global)
+    {
+        const auto has0 = this->has<T>(keyword);
         const auto& field_data = this->try_get<T>(keyword).field_data();
 
         if (has0) {
-            if (global)
-                return this->global_copy(field_data.data, field_data.kw_info.scalar_init);
-            else
-                return field_data.data;
-        } else {
-            if (global) {
-                const auto& kw_info = Fieldprops::keywords::global_kw_info<T>(keyword);
-                return this->global_copy(this->extract<T>(keyword), kw_info.scalar_init);
-            } else
-                return this->extract<T>(keyword);
+            return global
+                ? this->global_copy(field_data.data, field_data.kw_info.scalar_init)
+                : field_data.data;
+        }
+        else if (! global) {
+            return this->extract<T>(keyword);
         }
-    }
 
+        const auto& kw_info = Fieldprops::keywords::global_kw_info<T>(keyword);
+        return this->global_copy(this->extract<T>(keyword), kw_info.scalar_init);
+    }
```

Comment:
This is harder to read than before. As we return in the if-branch anyway, just skipping the else might reduce confusion.
```
if (! global) {
            return this->extract<T>(keyword);
}
...
```
Personally I would even switch to `if(global)` as before.

Reply:
> This is harder to read than before.

I've introduced a couple of helper functions to reduce the level of nesting.